### ISSUE TITLE
Command Set onInit is called twice when ClientSideComponentProperties are set using Add-PnPCustomAction

### ISSUE THREAD

**Issue by AkiraSPO** [SharePoint/sp-dev-docs#8529]
### Target SharePoint environment

SharePoint Online

### What SharePoint development model, framework, SDK or API is this about?

💥 SharePoint Framework

### Developer environment

Windows

### What browser(s) / client(s) have you tested

- [ ] 💥 Internet Explorer
- [ ] 💥 Microsoft Edge
- [X] 💥 Google Chrome
- [ ] 💥 FireFox
- [ ] 💥 Safari
- [ ] mobile (iOS/iPadOS)
- [ ] mobile (Android)
- [ ] not applicable
- [ ] other (enter in the "Additional environment details" area below)

### Additional environment details

- browser version - Chrome Version 107.0.5304.62 (Official Build) (64-bit)
- SPFx version - 1.9.1
- Node.js version - v10.24.0

### Describe the bug / error

- I created a simple ListView commandset extension by following this article https://learn.microsoft.com/en-us/sharepoint/dev/spfx/extensions/get-started/building-simple-cmdset-with-dialog-api

- And added  _console.log(this.properties.sampleTextOne);_ to the _onInit()_ function
- Deployed this component to the AppCatalog of a test site.
- I then used the following command to give specific values for ClientSideComponentProperties on a particular site:

_Add-PnPCustomAction -ClientSideComponentId "{_clientsidecomponentId from element.xml_ }"  -Title "HelloWorld" -Name "command-extension-client-side-solution" -Location ClientSideExtension.ListViewCommandSet.CommandBar -ClientSideComponentProperties: '{"sampleTextOne":"CommandText1","sampleTextTwo":"CommandText2"}' -Scope Site_

When I opened a SharePoint list that has this command bar items, here are a few observations that look like an issue:
1. Till the time I did not add ClientSideComponentProperties using 'Add-PnPCustomAction' the onInit() was getting called only once and was picking up values from element.XML, as expected.
2. When I added ClientSideComponentProperties using the pnp command,  I could see the onInit function being called twice (console.log() was logging 2 messages).
3. When onInit() was called twice, it was picking-up the properties set using pnp command for the first time and then picked up the values from element.xml the second time it got executed.
4.  If I remove the ClientSideComponentProperties using Remove-PnPCustomAction, onInit() was again getting called only once.
5. I tried completely removing the <ClientSideComponentProperties > from element.XML and then the Init() function returned 'undefined' for the property values, which means it did not read the properties set via pnp command at all.

### Steps to reproduce

1. Create a list view command set extension using this article: https://learn.microsoft.com/en-us/sharepoint/dev/spfx/extensions/get-started/building-simple-cmdset-with-dialog-api
2. Add console.log(this.properties.sampleTextOne); before the return statement in override
  public onInit(): Promise<void> {} function
3. Build and deploy the solution to the app catalog of a test site.
4. Add ClientSideComponentProperties using following command:
 _Add-PnPCustomAction -ClientSideComponentId "{_clientsidecomponentId from element.xml_ }"  -Title "HelloWorld" -Name "command-extension-client-side-solution" -Location ClientSideExtension.ListViewCommandSet.CommandBar -ClientSideComponentProperties: '{"sampleTextOne":"CommandText1","sampleTextTwo":"CommandText2"}' -Scope Site_

5. Open a list that has this command set extension available in command bar and open Developer tools > Console
6. You should see the following 2 messages logged in console, both coming from onInit() function. 

> CommandText1

> One item is selected in the list.

### Expected behavior

I expect the code to run onInit() just once and use property values defined using PnpCustom action command for the given site.

**Comment by AJIXuMuK** [comment]
@AkiraSPO  - thanks for reporting this one.
Few questions:
- is your solution tenant-wide deployed?
- You said there is an `elements.xml` in the solution, meaning that the command set is added to the list automatically when installed, correct?
- what `Location` value do you have in `elements.xml`?

It seems to me that `Add-PnPCustomAction` command adds a second copy of the command set to the list, not changing the properties of the existing one.

If you just need to update the properties, you'd probably need to `Get-PnPCustomAction` and update the returned object instead of adding a new copy.

**Comment by AkiraSPO** [comment]
> 

Thank  you for looking into the issue, please find below responses:
is your solution tenant-wide deployed? - __No, I am testing this list view command set only on my test site. But I do have an_ ApplicationCustomizer where I am observing the same behavior_
You said there is an elements.xml in the solution, meaning that the command set is added to the list automatically when installed, correct? - _Yes, that's correct._
what Location value do you have in elements.xml? - _Location="ClientSideExtension.ListViewCommandSet.CommandBar"_

You maybe right about the adding vs updating properties. I'll try the 'update' way and let you know if that works.

**Comment by AkiraSPO** [comment]
@AJIXuMuK  - you are right about Get-PnPCustomAction and updating the returned object.  Add-PnPCustomAction  was indeed creating a second copy of properties. In all my pnp commands, I was scoping them to Site  (-Scope Site) but it looks like when I deployed the code to a particular site app catalog, the properties were added with 'Web' scope. So the onInit() was first running for the Site scoped properties that the Add-PnPCustomAction added and then the second time for the web scoped properties that the code added. 
I am still not sure why SharePoint would allow the same Component GUID to be deployed at Web and Site scopes and then run onInit() for each one, but your suggestion has solved my current problem, so Thank you!
